### PR TITLE
dev-python/ipython: bump to 7.14.0; bug #691294

### PR DESCRIPTION
**Pull request by blinkingtwelve**
Diff with `ipython-7.5.0.ebuild`:

```
--- ipython-7.5.0.ebuild	2020-05-07 09:32:45.302702253 +0200
+++ ipython-7.14.0.ebuild	2020-05-07 10:11:55.144347786 +0200
@@ -24,8 +24,8 @@
 	dev-python/jedi[${PYTHON_USEDEP}]
 	dev-python/pexpect[${PYTHON_USEDEP}]
 	dev-python/pickleshare[${PYTHON_USEDEP}]
-	>=dev-python/prompt_toolkit-2[${PYTHON_USEDEP}]
-	<dev-python/prompt_toolkit-2.1[${PYTHON_USEDEP}]
+	>=dev-python/prompt_toolkit-3.0.2[${PYTHON_USEDEP}]
+	<dev-python/prompt_toolkit-3.1[${PYTHON_USEDEP}]
 	dev-python/pygments[${PYTHON_USEDEP}]
 	dev-python/traitlets[${PYTHON_USEDEP}]
 	matplotlib? ( dev-python/matplotlib[${PYTHON_USEDEP}] )
```

This was based on changed requirements per https://github.com/ipython/ipython/blob/7.14.0/setup.py#L193 . I've lightly tested with prompt_toolkit v3, and not with v2, hence I didn't include v2 in the deps even though it seems (per [requirements](https://github.com/ipython/ipython/blob/7.14.0/setup.py#L193)) that ipython-7.14.0 should work with v2 as well with v3. Moving forward etc :sun_behind_small_cloud: .

Closes: https://bugs.gentoo.org/691294
Signed-off-by: blinkingtwelve <blinkingtwelve@gavagai.eu>

### CODE REVIEW

**Comment by mgorny** [comment]
I'll give it a spin. I see the package would use some more fixes but nothing urgent or worth your time.

**Comment by boerwastaken** [comment]
Thanks, ran the tests, found the culprit (network-accessing test), and disabled it perusing your guide.

**Comment by blinkingtwelve** [comment]
Thanks, ran the tests, found the culprit (network-accessing test), and disabled it perusing your guide.

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 1]
```diff
@@ -91,6 +91,7 @@ src_test() {
 
 python_test() {
 	distutils_install_for_testing
+	sed -i 's/^def \(test_embed_svg_url():$\)/def _\1/g' build/test/lib/IPython/core/tests/test_display.py || die
```
Comment (mgorny) [inline dev-python/ipython/ipython-7.14.0.ebuild:94]: Just edit the source file in `prepare_all`, also please wrap lines.

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+PYTHON_REQ_USE='readline,sqlite,threads(+)'
+
+inherit distutils-r1 eutils virtualx
+
+DESCRIPTION="Advanced interactive shell for Python"
+HOMEPAGE="http://ipython.org/"
```
Comment (charIes17) [inline dev-python/ipython/ipython-7.14.0.ebuild:12]: https://ipython.org/

Comment (boerwastaken) [reply dev-python/ipython/ipython-7.14.0.ebuild]: Fixed

File: dev-python/ipython/ipython-7.14.0.ebuild (hunk 1 above)
Comment (boerwastaken) [reply dev-python/ipython/ipython-7.14.0.ebuild]: Fixed

**Comment by boerwastaken** [comment]
> Your sign-off needs to contain your real legal name.

Fixed

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+inherit distutils-r1 eutils virtualx
+
+DESCRIPTION="Advanced interactive shell for Python"
+HOMEPAGE="https://ipython.org/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="amd64 ~arm ~arm64 ~ppc ppc64 x86 ~amd64-linux ~x86-linux"
```
Comment (mgorny) [inline dev-python/ipython/ipython-7.14.0.ebuild:17]: You don't bump straight to stable.

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+	dev-python/pygments[${PYTHON_USEDEP}]
+	dev-python/traitlets[${PYTHON_USEDEP}]
+	matplotlib? ( dev-python/matplotlib[${PYTHON_USEDEP}] )
+"
+
+RDEPEND="${CDEPEND}
+	nbconvert? ( dev-python/nbconvert[${PYTHON_USEDEP}] )"
+
+DEPEND="${CDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/ipython/ipython-7.14.0.ebuild:38]: This is redundant now.

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	<dev-python/prompt_toolkit-3.1[${PYTHON_USEDEP}]
+	dev-python/pygments[${PYTHON_USEDEP}]
+	dev-python/traitlets[${PYTHON_USEDEP}]
+	matplotlib? ( dev-python/matplotlib[${PYTHON_USEDEP}] )
+"
+
+RDEPEND="${CDEPEND}
+	nbconvert? ( dev-python/nbconvert[${PYTHON_USEDEP}] )"
+
+DEPEND="${CDEPEND}
```
Comment (mgorny) [inline dev-python/ipython/ipython-7.14.0.ebuild:37]: These go to BDEPEND in EAPI 7. Also, I honestly doubt you need this 'CDEPEND' outside `test?`

File: dev-python/ipython/ipython-7.14.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+	)
+	qt5? ( dev-python/qtconsole[${PYTHON_USEDEP}] )
+	smp? (
+		>=dev-python/ipykernel-5.1.0[${PYTHON_USEDEP}]
+		>=dev-python/ipyparallel-6.2.3[${PYTHON_USEDEP}]
+	)"
+
+PATCHES=( "${FILESDIR}"/2.1.0-substitute-files.patch )
+
+DISTUTILS_IN_SOURCE_BUILD=1
```
Comment (mgorny) [inline dev-python/ipython/ipython-7.14.0.ebuild:66]: Is this still necessary?

**Comment by juippis** [comment]
fc844c74728bab3f05afe7c0375e61bae610e6cb

**Pull request closed without merging**